app/tasks/maintenance/t20250804destroy_dossier_depose_notification_of_groupe_instructeur_task.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Maintenance
+  class T20250804destroyDossierDeposeNotificationOfGroupeInstructeurTask < MaintenanceTasks::Task
+    # Documentation: cette tâche permet de supprimer toutes les notification de
+    # type dossier_depose qui sont rattachées à un groupe instructeur
+
+    include RunnableOnDeployConcern
+    include StatementsHelpersConcern
+
+    # Uncomment only if this task MUST run imperatively on its first deployment.
+    # If possible, leave commented for manual execution later.
+    # run_on_first_deploy
+
+    def collection
+      DossierNotification
+        .where(notification_type: :dossier_depose)
+        .where(instructeur_id: nil)
+        .in_batches
+    end
+
+    def process(batch_of_notifications)
+      batch_of_notifications.delete_all
```

Comment:
est ce que ça aurait un intérêt de regrouper les 2 MT en une ?

ici pour chacune des  anciennes notifications on pourrait faire un truc du genre :

```
notification.transaction do
  dossier_id = notification.dossier_id
  display_at = notification.display_at
  instructeur_ids = notification.groupe_instructeur.instructeur_ids

  to_create = instructeur_ids.map { |instructeur_id| { dossier_id:, notification_type:, instructeur_id:, display_at: } }
  DossierNotification.create(to_create)
  notification.delete
end
```

Reply:
a ce putain d'offest. peut etre que t'as raison.

Reply:
en fait y a pas tjs un pb d'offset car si la collection renvoyée est une relation active record, MT itère sur les batchs en faisant `…where id > 12345 order by id LIMIT 1000`  où  12345 est le dernier id traité du précédent batch. (c'était peut-être différent dans une version plus ancienne de rails, ou si le modèle sous-jacent ne permet pas de s'appuyer sur les ids).